### DISCUSSION TITLE
Steady-State Problem on a Single Element

### DISCUSSION THREAD

**Discussion by smpeyres** [idaholab/moose#24730] [Q&A General] [answered]
I am trying to implement a steady-state, zero-dimensional chemical kinetics model using CRANE. Traditionally, an ODETimeDerivative kernel is added for each of the species and each species' density is initialized. The ChemicalReactions block parses out the listed reactions as individual reaction kernels, which are quite similar to MOOSE's native & relatively simple Reaction kernels. 

I tried to implement a steady-solve for a two-reaction network for which the [steady-state solution from a Transient solve is known and verified](https://crane-plasma-chemistry.readthedocs.io/en/latest/TwoReactionArgon.html). However, my initial attempts result in the electron density 'e' plummeting many orders of magnitude and the ion density 'Ar+' remaining constant. Both of these should change but remain equal to each other. 

I believe my issue is one of applying ill-posed constraints on the values. The reaction-network used has a steady-state solution for 'e' = 'Ar+' that is equal to the ratio of the two rate coefficients with no dependence on the value of 'Ar'. A simple steady-state problem using diffusion on a 1D mesh seems to work just fine because the boundary conditions seem to provide sufficient constraints. 

You can find the input file and associated data in [my fork of crane -> branch : steady -> problems/steady/TwoReactionArgon](https://github.com/smpeyres/crane/tree/steady/problems/steady/TwoReactionArgon).

Any suggestions? 

Thank you!

**Comment by GiudGiud** [comment, marked as answer]
Hello

I wont be able to help but I would recommend creating a github repo and hosting the files instead of sharing a zip file. Not too many people will download a zip file from the internet.

@mtnwalker for comments on the physics

Guillaume

**Reply by smpeyres** [reply]
Yes, I will do that! I'll update the post with a link to the branch when I am finished.

**Reply by smpeyres** [reply]
Update: I got it working on the branch linked with draft 4. Thank you!

**Comment by GiudGiud** [comment]
@tanoret @mtnwalker who can we tag on chemical reactions with MOOSE?
